service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/ManageDocumentsUploadedEventHandler.java

```diff
@@ -0,0 +1,341 @@
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static uk.gov.hmcts.reform.fpl.enums.cfv.DocumentType.CASE_SUMMARY;
+import static uk.gov.hmcts.reform.fpl.enums.cfv.DocumentType.COURT_BUNDLE;
+import static uk.gov.hmcts.reform.fpl.enums.cfv.DocumentType.COURT_CORRESPONDENCE;
+import static uk.gov.hmcts.reform.fpl.enums.cfv.DocumentType.POSITION_STATEMENTS;
+import static uk.gov.hmcts.reform.fpl.enums.cfv.DocumentType.SKELETON_ARGUMENTS;
+import static uk.gov.hmcts.reform.fpl.enums.notification.DocumentUploaderType.SOLICITOR;
+import static uk.gov.hmcts.reform.fpl.service.docmosis.DocumentConversionService.PDF;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.DATE;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateBaseUsingFormat;
+import static uk.gov.hmcts.reform.fpl.utils.DocumentsHelper.hasExtension;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ManageDocumentsUploadedEventHandler {
+
+    private final CafcassLookupConfiguration cafcassLookupConfiguration;
+    private final CafcassNotificationService cafcassNotificationService;
+    private final FurtherEvidenceNotificationService furtherEvidenceNotificationService;
+    private final SendDocumentService sendDocumentService;
+    private final TranslationRequestService translationRequestService;
+
+    private static final String LIST = "•";
+    public static final String FURTHER_DOCUMENTS_FOR_MAIN_APPLICATION = "Further documents for main application";
+    public static final String CORRESPONDENCE = "Correspondence";
+
+    /**
+     * Base on the confidential level configuration, send notification to the given recipient.
+     * e.g.
+     * Given: A CTSC level doc and a LA level doc are uploaded
+     * Case 1: Both files are allowed to be sent to a CTSC level user
+     * Case 2: Only LA level doc is allowed to be sent to a LA level user
+     * Case 2: No doc is allowed to be sent to a non-confidential level user
+     *
+     * @param event the ManageDocumentsUploadedEvent
+     */
+    @Async
+    @EventListener
+    public void sendDocumentsUploadedNotification(final ManageDocumentsUploadedEvent event) {
+        buildConfigurationMapGroupedByRecipient(event)
+            .forEach((recipients, getConfidentialLevelFunction) -> {
+                if (isNotEmpty(recipients)) {
+                    List<Element<NotifyDocumentUploaded>> documentsToBeSent =
+                        consolidateMapByConfiguration(event, getConfidentialLevelFunction)
+                            .entrySet().stream()
+                            .map(Map.Entry::getValue)
```

Comment:
`.values().stream()` instead of `.entrySet().stream().

Reply:
updated